### PR TITLE
fix(patch_obligation): Fix patch request bugs in obligation

### PR DESCRIPTION
**Pull request by deo002**
When we unmarshal json, the undefined keys take zero values in structs. So, there is no way to differentiate between an undefined value and an actual zero value when it is passed in json in a PATCH request.
Example: 
{
    "active": false,
    "text": "Lorem ipsum"
} 
and 
{
    "text": "Lorem ipsum"
} 
Structs of both these unmarshalled json objects will have active as false.

### CODE REVIEW

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Review by avinal — APPROVED** [review]
Looks good to me

File: pkg/models/types.go [hunk 1]
```diff
@@ -246,42 +246,39 @@ type AuditResponse struct {
...
 // Obligation represents an obligation record in the database.
 type Obligation struct {
-	Id             int64  `json:"id" gorm:"primary_key" example:"147"`
-	Topic          string `json:"topic" gorm:"unique" example:"copyleft"`
-	Type           string `json:"type" enums:"obligation,restriction,risk,right"`
+	Id             int64  `gorm:"primary_key" json:"id" example:"147"`
+	Topic          string `gorm:"unique" json:"topic" example:"copyleft"`
+	Type           string `json:"type" example:"risk"`
 	Text           string `json:"text" example:"Source code be made available when distributing the software."`
-	Classification string `json:"classification" enums:"green,white,yellow,red"`
```
Comment (GMishx) [inline pkg/models/types.go:253]: Kindly revert the removal of `enums` as it is used by OpenAPI documentation.

**Review by GMishx — APPROVED** [review]
Tested, working as expected.

**Pull request merged**